### PR TITLE
Add support for 'room list' and 'room occupants <room>'

### PR DESCRIPTION
**Pull request by KimNorgaard**
This is using the REST API. It's not ideal since the REST API is quite limited. But it might be useful to plugins.

I also made a class for the GHC API. This should make calls to the API a bit more uniform.

### CODE REVIEW

File: hangouts_chat.py [hunk 1]
```diff
@@ -13,47 +14,124 @@
...
+    @property
+    def client(self):
+        return self.credentials.authorize(httplib2.Http())
+
+    def _get(self, uri: str, query_string: str = None) -> Optional[dict]:
+        url = '{}/{}'.format(self.base_url, uri)
+        if query_string:
+            url += '?{}'.format(query_string)
+        result, content = self.client.request(uri=url, method='GET')
+        if result['status'] == "200":
```
Comment (sinkingpoint) [inline hangouts_chat.py]: Nitpick: consistent use of ' vs " here

File: hangouts_chat.py [hunk 2]
```diff
@@ -13,47 +14,124 @@
...
+    def _post(self, uri: str, body: dict, query_string: str = None) -> Optional[dict]:
+        url = '{}/{}'.format(self.base_url, uri)
+        if query_string:
+            url += '?{}'.format(query_string)
+        result, content = self.client.request(
+            uri=url,
+            method='POST',
+            headers={'Content-Type': 'application/json; charset=UTF-8'},
+            body=json.dumps(body))
+        if result['status'] == "200":
```
Comment (sinkingpoint) [inline hangouts_chat.py]: This logic feels repeated. Could we split it out to a generic `_request` method?

File: hangouts_chat.py [hunk 3]
```diff
@@ -213,7 +299,12 @@ def mode(self):
...
     def query_room(self, room):
-        return HangoutsChatRoom(room, self.creds_file)
+        return HangoutsChatRoom(room, self.chat_api)
 
     def rooms(self):
-        return None
+        spaces = self.chat_api.get_spaces()
+        rooms = ['{} ({})'.format(space['displayName'], space['name'])
+                 for space in list(spaces) if space['type'] == 'ROOM']
+        log.debug(rooms)
```
Comment (sinkingpoint) [inline hangouts_chat.py]: Leftover debugging statement?

File: hangouts_chat.py (hunk 1 above)
Comment (KimNorgaard) [reply hangouts_chat.py]: Fixed with latest commit

File: hangouts_chat.py (hunk 2 above)
Comment (KimNorgaard) [reply hangouts_chat.py]: Done :)

File: hangouts_chat.py (hunk 3 above)
Comment (KimNorgaard) [reply hangouts_chat.py]: Removed

**Comment by KimNorgaard** [comment]
Thanks for the review.

**Pull request merged**